include/cupla/device/Atomic.hpp

```diff
@@ -0,0 +1,176 @@
+        void functionName(                                                     \
+            T_Acc const & acc,                                                 \
+            T_Type *ptr,                                                       \
+            T_Type const & value                                               \
+        )                                                                      \
+        {                                                                      \
+            ::alpaka::atomic::atomicOp< alpakaOp >(                            \
+                acc,                                                           \
+                ptr,                                                           \
+                value,                                                         \
+                T_Hierarchy{}                                                  \
+            );                                                                 \
+        }                                                                      \
+                                                                               \
+        /*! @param hierarchy hierarchy level within the operation is atomic    \
+         */                                                                    \
+        template<                                                              \
+            typename T_Acc,                                                    \
+            typename T_Type,                                                   \
+            typename T_Hierarchy = alpaka::hierarchy::Grids                    \
+        >                                                                      \
+        ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE                                    \
+        void functionName(                                                     \
+            T_Acc const & acc,                                                 \
+            T_Type *ptr,                                                       \
+            T_Type const & value,                                              \
+            T_Hierarchy const & hierarchy = T_Hierarchy()                      \
+        )                                                                      \
+        {                                                                      \
+            functionName< T_Hierarchy >(                                       \
+                acc,                                                           \
+                ptr,                                                           \
+                value                                                          \
+            );                                                                 \
+        }                                                                      \
+        /*!@}                                                                  \
+         */
+
+        /// atomic addition
```

Comment:
Perhaps add a doxygen group for these atomic operations?

Reply:
The macro is providing the grouped doxygen docu. This line is the brief.

Reply:
I thought the grouping ended just 2 lines up from here?